Explain better how dependencies work with mqtt example (#1768)
home-assistant/developers.home-assistant: docs/creating_integration_manifest.md

## Dependencies

> Dependencies are other Home Assistant integrations that you want Home Assistant to set up successfully prior to the integration being loaded. This can be necessary in case you want to offer functionality from that other integration, like using webhooks or a MQTT connection. Adding an integration to dependencies will ensure the depending integration is loaded before setup, but it does not guarantee all configuration entries have been set up. See the [MQTT section](#mqtt) for more details on handling this for MQTT.

Reviewer: ```suggestion
Dependencies are other Home Assistant integrations you want Home Assistant to set up successfully before the integration is loaded. Adding an integration to dependencies will ensure the depending integration is loaded before setup, but it does not guarantee all dependency configuration entries have been set up. Adding dependencies can be necessary if you want to offer functionality from that other integration, like webhooks or an MQTT connection. See the [MQTT section](#mqtt) for more details on handling this for MQTT.
```

> Dependencies are other Home Assistant integrations you want Home Assistant to set up successfully before the integration is loaded. Adding an integration to dependencies will ensure the depending integration is loaded before setup, but it does not guarantee all dependency configuration entries have been set up. Adding dependencies can be necessary if you want to offer functionality from that other integration, like webhooks or an MQTT connection. Adding an [after dependency](#after-dependencies) might be a better alternative if a dependency is optional but not critical. See the [MQTT section](#mqtt) for more details on handling this for MQTT.

Reviewer: Related code for anyone looking at this 

https://github.com/home-assistant/core/blob/41515249a0ba3a4a24e3fe55dd2c9a9dc46a798f/homeassistant/setup.py#L295

https://github.com/home-assistant/core/blob/95023ecf08f01b5ebb88fb0dab285263dd435439/homeassistant/setup.py#L125

}
```

If your integration requires `mqtt`, make sure it is added to the [dependencies](#dependencies).

> The MQTT integration for example does not implement `async_setup` and the config entry is set up via `async_setup_entry`. This also applies for integrations that depend on MQTT and that have implemented a platform setup only via `async_setup_platform`. Depending integrations should wait using `await mqtt.async_wait_for_mqtt_client(hass)` for the MQTT client to become available before they can subscribe. The `async_wait_for_mqtt_client` method will block and return `True` till the MQTT client is available. If a dependency is optional but not critical, then adding an [after dependency](#after-dependencies) might be a better alternative.

Reviewer: ```suggestion

Integrations depending on MQTT should wait using `await mqtt.async_wait_for_mqtt_client(hass)` for the MQTT client to become available before they can subscribe. The `async_wait_for_mqtt_client` method will block and return `True` till the MQTT client is available.
```

Reviewer: I think `Adding an [after dependency](#after-dependencies) might be a better alternative if a dependency is optional but not critical.` belongs above ?
